src/core/reference/include/openvino/reference/round.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cfenv>
+#include <cmath>
+#include <type_traits>
+
+#include "openvino/op/round.hpp"
+
+namespace ov {
+template <class T>
+constexpr bool is_floating_point() {
+    using U = typename std::decay<T>::type;
+    return std::is_floating_point<U>::value || std::is_same<float16, U>::value || std::is_same<bfloat16, U>::value;
+}
+
+namespace reference {
+void set_round_mode(const op::v5::Round::RoundMode& mode);
+
+/**
+ * @brief Rounding algorithm for ov::op::v5::Round::RoundMode::HALF_TO_EVEN.
+ *
+ * @tparam T     Value type.
+ * @param value  Value for rounding.
+ * @return       Rounded value.
+ */
+template <typename T>
+T round_to_nearest_even(T value) {
+    return std::nearbyint(value);
+}
+
+/**
+ * @brief Rounding algorithm for ov::op::v5::Round::RoundMode::HALF_AWAY_FROM_ZERO.
+ *
+ * @tparam T     Value type.
+ * @param value  Value for rounding.
+ * @return       Rounded value.
+ */
+template <typename T>
+T round_half_away_zero(T value) {
+    return std::round(value);
+}
+
+/**
+ * @brief Reference implementation of Round operator.
+ *
+ * Used when T is OpenVINO floating type.
+ *
+ * @param arg    Input tensor with data to round.
+ * @param out    Output tensor with rounding result.
+ * @param count  Number of elements in input tensor.
+ * @param mode   Rounding mode.
+ */
+template <typename T, typename std::enable_if<ov::is_floating_point<T>()>::type* = nullptr>
+void round(const T* arg, T* out, const size_t count, const op::v5::Round::RoundMode& mode) {
+    set_round_mode(mode);
```

Comment:
Should we restore previous round mode at the end of this function?

Reply:
Yes, done as simple guard which can be reused in other reference implementations which change rounding direction.